GoMoney/Scences/Detail/DetailViewController.swift

```diff
@@ -0,0 +1,122 @@
+import UIKit
+
+class DetailViewController: GMMainViewController {
+    // MARK: - Properties
+
+    let viewModel = DetailViewModel()
+    
+    lazy var detailView = DetailView(transaction: transaction)
+    
+    lazy var editButton = GMFloatingButton(
+        image: UIImage(systemName: "pencil")?.color(.white),
+        text: "Edit")
+    { [weak self] in
+        self?.editTransaction()
+    }
+    
+    var transaction: Expense? {
+        didSet {
+            guard let transaction = transaction else {
+                return
+            }
+            viewModel.transaction = transaction
+            detailView.transaction = transaction
+        }
+    }
+    
+    // MARK: - LifeCircle
+    
+    override func configureBackButton() {
+        super.configureBackButton()
+
+        navigationItem.rightBarButtonItems = [
+            UIBarButtonItem(image: UIImage(systemName: "personalhotspot")?.color(.white),
+                            style: .done,
+                            target: self,
+                            action: #selector(shareTransaction)),
+            
+            UIBarButtonItem(image: UIImage(systemName: "trash")?.color(.white),
+                            style: .done,
+                            target: self,
+                            action: #selector(deleteTransaction))
+        ]
+    }
+
+    override func setupLayout() {
+        title = "Detail"
+        super.setupLayout()
+        
+        view.addSubviews(detailView, editButton)
+        
+        detailView.anchor(
+            top: view.safeAreaLayoutGuide.topAnchor,
+            left: view.leftAnchor,
+            bottom: view.safeAreaLayoutGuide.bottomAnchor,
+            right: view.rightAnchor,
+            paddingTop: 32,
+            paddingLeft: 16,
+            paddingRight: 16)
+        
+        editButton.anchor(
+            bottom: view.safeAreaLayoutGuide.bottomAnchor,
+            right: view.rightAnchor,
+            paddingRight: 16)
+    }
+    
+    // MARK: Functions
+    
+    private func editTransaction() {
+        guard transaction != nil else {
+            alert(
+                title: "Error",
+                message: "Can't edit transaction",
+                actionTitle: "Cancel")
+            return
+        }
+        
+        let vc = EditViewController()
+        vc.transaction = transaction
+        vc.onApply = { newTrans in
+            self.detailView.transaction = newTrans
```

Comment:
weak self

Reply:
I think no need weak self here because `DetailViewController` doesn't point to `EditViewController`
I also checked by debug tool